lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -230,394 +227,449 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
+        <%= render_slot(col) %>
+      </div>
+    </div>
+    """
   end
 
-  @doc """
-  Renders a row component.
-
-  It can be rendered in any dashboard page via the `render_page/1` function:
-
-      def render_page(assigns) do
-        row(
-          components: [
-            card(...),
-            columns(...)
-          ]
-        )
-      end
-
-  You can see it in use the Home page and OS Data pages.
-
-  # Options
+  defp row_validate_columns_length(assigns) do
+    columns_length = length(assigns[:col] || [])
 
-  These are the options supported by the component:
-
-    * `:components` - Required. A list of components.
-      It can receive up to 3 components.
-      Each element will be one column.
-  """
-  @spec row(keyword()) :: component()
-  def row(assigns) do
-    assigns =
-      assigns
-      |> Map.new()
-      |> RowComponent.normalize_params()
-
-    {RowComponent, assigns}
+    if columns_length > 0 and columns_length < 4 do
+      assign(assigns, :columns_class, div(12, columns_length))
+    else
+      raise ArgumentError,
+            "row component must have at least 1 and at most 3 :col, got: " <>
+              inspect(columns_length)
+    end
   end
 
   @doc """
-  Renders a usage card component.
-
-  It can be rendered in any dashboard page via the `render_page/1` function:
-
-      def render_page(assigns) do
-        usage_card(
-          usages: [
-            %{
-              current: 10,
-              limit: 150,
-              dom_sub_id: "1",
-              title: "Memory",
-              percent: "13"
-            }
-          ],
-          dom_id: "memory"
-        )
-      end
+  Usage card component.
 
   You can see it in use the Home page and OS Data pages.
-
-  # Options
-
-  These are the options supported by the component:
-
-    * `:usages` - Required. A list of `Map` with the following keys:
-      * `:current` - Required. The current value of the usage.
-      * `:limit` - Required. The max value of usage.
-      * `:dom_sub_id` - Required. An unique identifier for the usage that will be concatenated to `dom_id`.
-      * `:percent` - The used percent if the usage. Default: `nil`.
-      * `:title` - Required. The title of the usage.
-      * `:hint` - A textual hint to show close to the usage title. Default: `nil`.
-
-    * `:dom_id` - Required. A unique identifier for all usages in this card.
-    * `:title` - The title of the card. Default: `nil`.
-    * `:hint` - A textual hint to show close to the card title. Default: `nil`.
   """
-  @spec usage_card(keyword()) :: component()
+  attr :title, :string, default: nil, doc: "The title above the card."
+  attr :hint, :string, default: nil, doc: "A textual hint to show close to the title."
+  attr :dom_id, :string, required: true, doc: "A unique identifier for all usages in this card."
+  attr :csp_nonces, :any, default: %{img: nil, script: nil, style: nil}, doc: "TODO!!"
```

Comment:
TODO!

Reply:
I think we have the documentation for them somewhere. :)

Reply:
```suggestion
  attr :csp_nonces, :any, required: true, default: %{img: nil, script: nil, style: nil},
    doc: "A copy of CSP nonces (`@csp_nonces`) used to render the page safely"
```

Note I am even making it required above. This is because I would argue you should always remember to pass them so they are rendered properly!

Reply:
Yeah, I did the same but the compiler doesn't like it :P 

> Only one of :required or :default must be given

So what do you prefer?

Reply:
`:required`!